pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -1003,6 +1003,7 @@ func newNetworkAddonsForCR(cr *hcov1alpha1.HyperConverged, namespace string) *ne
 			LinuxBridge: &networkaddonsv1alpha1.LinuxBridge{},
 			Ovs:         &networkaddonsv1alpha1.Ovs{},
 			NMState:     &networkaddonsv1alpha1.NMState{},
+			KubeMacPool: &networkaddonsv1alpha1.KubeMacPool{},
```

Comment:
This is probably more related to line 1044; now we have a flag (`r.upgradeMode`) to signal that we are in the middle of an upgrade.
Do we have to reconcile this config map continuously or should we coordinate this with the upgrade process?

Reply:
When we add this attribute, it will render the status of CNAO "Progressing". Would that ensure that we signal the upgrade?

Reply:
It's detected by HCO comparing the value of a new environmental variable read from the CSV with what was stored on HCO CR in the previous run.
We are not going to reset it until all the observed components reports Available=true, Degraded=false, Progressing=False.
With https://github.com/kubevirt/hyperconverged-cluster-operator/pull/584 we also started explicitly checking the version reported by each single operator (still not all of them).

While with `r.upgradeMode=true`, HCO is not going to report true with the readiness probe so that OLM, and other users, should see it less flappy.

Reply:
I'm still not sure I follow. Is there any extra action needed to support this PR?

Reply:
OLM is not transactional and doesn't have any rollback mechanism.
The question is basically if directly changing this config map, regardless of the upgrade status of CNAO, is safe enough or if it can cause more troubles to an older version of CNAO that for, any reason, is still not completely progressed to a newer version.

Reply:
I see. It should be able to progress to a healthy state. However, we don't want to consume KubeMacPool from an old version of CNAO. Can we avoid changing this if we see the upgrade is not completed yet?

Reply:
yes, this is exactly what I'm suggesting.
HCO will set `r.upgradeMode=true` as soon as it detects that it's running with a version newer/different from what we tracked in the CSV.
Then we check the upgrade status of CNAO on line 1060; we can consume this there.

Reply:
I think I get it now. I'm not sure what would be the best solution, but saying "not reconfigure while upgrading" sounds like a safe bet.